tripy/tripy/frontend/trace/ops/reshape.py

```diff
@@ -80,7 +80,9 @@ def reshape_impl(
     },
     dtype_constraints={"input": "T1", "shape": "T2", constraints.RETURN_VALUE: "T1"},
 )
-def reshape(input: "tripy.Tensor", shape: Union["tripy.Shape", Sequence[Union[int, "tripy.Tensor"]]]) -> "tripy.Tensor":
+def reshape(
+    input: "tripy.Tensor", shape: Union["tripy.Shape", "tripy.Tensor", Sequence[Union[int, "tripy.Tensor"]]]
```

Comment:
Should it be `Union["tripy.Shape", "tripy.ScalarShape", Sequence[Union[int, "tripy.Tensor"]]]`?

Reply:
Also, it should be a sequence of shapescalars and not tensors right?

Reply:
And there are few more places to update, search for `Sequence[Union[int, "tripy.Tensor"]]]` in the codebase:
- fill
- expand
- ones, zeros
- iota
- reshape

Reply:
It can be Tensor also and not necessarily `ShapeScalar`. Yes, the Sequence should be updated to `ShapeScalar`.

Reply:
It can only be a tensor once we have DDS right? Until then it must be a shape of some kind